backend/main.py

```
@@ -9,46 +1,343 @@
+        sources: List[Dict[str, Any]] = []
+
+        return QueryResponse(
+            response=response_text,
+            sources=sources,
+            document_stats=stats if request.include_sources else {},
+            model_used=rag_system.get_model_name(),
+        )
+
+    except Exception as e:
+        logger.error(f"❌ Query failed: {e}")
+        raise HTTPException(status_code=500, detail=f"Query processing failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)